lib/runtime/newWasmer/memory_test.go

```diff
@@ -0,0 +1,130 @@
+package newWasmer
+
+import (
+	"testing"
+	"unsafe"
+
+	"github.com/stretchr/testify/require"
+	"github.com/wasmerio/wasmer-go/wasmer"
+)
+
+func createInstance(t *testing.T) (*wasmer.Instance, error) {
+	t.Helper()
+	// We are using the text representation of the module here. Taken from:
+	// https://github.com/wasmerio/wasmer-go/blob/23d786b6b81ad93e2b974d2f4510bea374f0f37c/examples/example_memory_test.go#L28
+	wasmBytes := []byte(`
+		(module
+		  (type $mem_size_t (func (result i32)))
+		  (type $get_at_t (func (param i32) (result i32)))
+		  (type $set_at_t (func (param i32) (param i32)))
+		  (memory $mem 1)
+		  (func $get_at (type $get_at_t) (param $idx i32) (result i32)
+		    (i32.load (local.get $idx)))
+		  (func $set_at (type $set_at_t) (param $idx i32) (param $val i32)
+		    (i32.store (local.get $idx) (local.get $val)))
+		  (func $mem_size (type $mem_size_t) (result i32)
+		    (memory.size))
+		  (export "get_at" (func $get_at))
+		  (export "set_at" (func $set_at))
+		  (export "mem_size" (func $mem_size))
+		  (export "memory" (memory $mem)))
+	`)
+
+	wasmerConfig := wasmer.NewConfig()
+	wasmerConfig.UseSinglepassCompiler()
```

Comment:
Can go over this in standup, but I think singlepass compiler is the way to go for us so thought I would use it here